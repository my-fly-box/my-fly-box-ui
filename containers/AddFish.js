import React, { Component } from "react";
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import { connect } from "react-redux";
import FishForm from '../components/FishForm'
import { clearFishEntry, addFish } from '../actions';
import { addFishToAPI } from "../ApiCalls";

class AddFish extends Component {
  constructor(props) {
    super(props);
  }

  addFish = () => {
    const currentFishEntry = this.props.currentFishEntry
    addFishToAPI(currentFishEntry)
      .then(data => {
        this.props.addFish(data.data);
        this.props.clearFishEntry();
      }
    )
  }

  render() {
    return (
      <View style={styles.container}>
        {/* working on getting camera integrated */}
        <TouchableOpacity style={styles.button}
            onPress = {() => {
              this.props.navigation.navigate('PhotoSelector');
              }}>
            <Text style={styles.button}>Upload Image</Text>
          </TouchableOpacity>

        <FishForm />
        
        <View style={styles.buttonContainer}>
          <TouchableOpacity style={styles.button}
            onPress = {() => {
              this.props.navigation.navigate('FishCaught');
              this.props.clearFishEntry();
              }}>
            <Text>Cancel</Text>
          </TouchableOpacity>
        
          <TouchableOpacity style={styles.button}
            onPress = {() => {
              this.addFish();
              this.props.navigation.navigate('FishCaught');
              this.props.clearFishEntry();
              }}>
            <Text>Submit</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 45,
    width: "100%",
    height:"100%",
    justifyContent: 'center',
  },
  buttonContainer: {
    width: "80%",
    height:"100%", 
    flex: 1, 
    flexDirection: 'row',
    alignSelf: "center",
  },
  button: {
    backgroundColor: 'white',
    color: 'white',
    flex: 1,
    alignSelf: "center",
    margin: 5,
    marginTop: "20%",
    height: "20%",
    width: "40%",
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 2,
    borderColor: '#2A9D8F',
    borderRadius: 3,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },
})

const mapStateToProps = state => ({
  currentFishEntry: state.currentFishEntry,
})

const mapDispatchToProps = (dispatch) => ({
  clearFishEntry: () => dispatch( clearFishEntry() ),
  addFish: data => dispatch( addFish(data) ),
})

export default connect(mapStateToProps, mapDispatchToProps)(AddFish);