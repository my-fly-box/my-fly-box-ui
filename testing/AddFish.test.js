import React from 'react';
import 'react-native'
import AddFish from '../containers/AddFish'
import renderer from 'react-test-renderer'

describe('AddFish', () => {
  it('renders the AddFish', () => {
    const snap = renderer.create(
      <AddFish />
    ).toJSON();
    expect(snap).toMatchSnapshot();
  })
})