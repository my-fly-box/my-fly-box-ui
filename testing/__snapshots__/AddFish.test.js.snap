// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFish renders the AddFish 1`] = `
<View>
  <Text>
    Add Fish
  </Text>
</View>
`;
